src/components/Controls.vue

```diff
@@ -1,10 +1,151 @@
+    newNode['ORIGIN'] = node['ORIGIN'];
+    newNode['ORIGIN_CITY'] = node['ORIGIN_CITY'];
+    newNode['ORIGIN_STATE'] = node['ORIGIN_STATE'];
+    newNode['_key'] = node['_key'];
+    newNode['_id'] = node['id'];
+    newNode['parent'] = undefined;
+    newNodes.push(newNode);
+  });
+
+  // Construct a supernode using the node interface and add some more fields
+  const californiaSuperNode: any = {};
+  californiaSuperNode['ORIGIN'] = 'CA';
+  californiaSuperNode['ORIGIN_STATE'] = 'California';
+  californiaSuperNode['_key'] = 'CA';
+  californiaSuperNode['_id'] = 'supernodes/CA';
+  californiaSuperNode['children'] = [];
+
+  // add the super node
+  newNodes.push(californiaSuperNode);
+
+  // add an index
+  for (let index = 0; index < newNodes.length; index++) {
+    newNodes[index]['index'] = index;
+  }
+
+  // find the index of the super node
+  const superIndex = newNodes.findIndex(
+    (node: any) => node['_id'] === 'supernodes/CA',
+  );
+
+  newNodes.forEach((node) => {
+    if (
+      node['_id'] !== 'supernodes/CA' &&
+      node['ORIGIN_STATE'] === 'California'
+    ) {
+      node['parent'] = superIndex;
+      newNodes[superIndex]['children'].push(node['index']);
+    }
+  });
+
+  // console.log('THE NEW NODE LIST');
+  // console.log(newNodes);
+
+  // node dictionary
+  const newNodeDict: any = {};
+  newNodes.forEach((node) => {
+    newNodeDict[node['_id']] = node['parent'];
+  });
+  // console.log(newNodeDict);
+
+  // construct new edges
+  const newLinks: any = [];
+  edges.forEach((link) => {
+    const newLink: any = {};
+    newLink['_key'] = link['_key'];
+    newLink['_from'] = link['source'];
+    newLink['_to'] = link['target'];
+    newLink['ORIGIN_STATE'] = link['ORIGIN_STATE'];
+    newLink['DEST_STATE'] = link['DEST_STATE'];
+    newLink['AIR_TIME_MEDIAN'] = link['AIR_TIME_MEDIAN'];
+    newLink['AIR_TIME_MEAN'] = link['AIR_TIME_MEAN'];
+    newLink['DEP_DELAY_MAX'] = link['DEP_DELAY_MAX'];
+    newLink['AIR_TIME_MAX'] = link['AIR_TIME_MAX'];
+    newLink['AIRLINE'] = link['AIRLINE'];
+    newLink['AIR_TIME_MIN'] = link['AIR_TIME_MIN'];
+    newLink['DEP_DELAY_MEDIAN'] = link['DEP_DELAY_MEDIAN'];
+    newLink['DEP_DELAY_MEAN'] = link['DEP_DELAY_MEAN'];
+    newLink['DEP_DELAY_MIN'] = link['DEP_DELAY_MIN'];
+    newLinks.push(newLink);
+  });
+
+  // update all the source and destinations
+  newLinks.forEach((link: any) => {
+    const linkFrom = link['_from'];
+    const linkTo = link['_to'];
+    if (newNodeDict[linkFrom] != undefined) {
+      const index = newNodeDict[linkFrom];
+      const newLinkFrom = newNodes[index]['_id'];
+      link['_from'] = newLinkFrom;
+    }
+    if (newNodeDict[linkTo] != undefined) {
+      const index = newNodeDict[linkTo];
+      const newLinkTo = newNodes[index]['_id'];
+      link['_to'] = newLinkTo;
+    }
+  });
+
+  // update the edge id of the links
+  for (let index = 0; index < newLinks.length; index++) {
+    const edgeName = 'edges/' + index;
```

Comment:
Use a template literal here:

```suggestion
    const edgeName = `edges/${index}`;
```